library/cipher.c

```diff
@@ -233,9 +234,11 @@ int mbedtls_cipher_set_iv( mbedtls_cipher_context_t *ctx,
         if( actual_iv_size > iv_len )
             return( MBEDTLS_ERR_CIPHER_BAD_INPUT_DATA );
     }
-
-    memcpy( ctx->iv, iv, actual_iv_size );
-    ctx->iv_size = actual_iv_size;
+    if ( actual_iv_size )
```

Comment:
I would prefer to have explicit checks if you would like to keep this `if` statement.

Reply:
Why is this `if` statement necessary?

Reply:
since I changed the default iv_size of `ECB` mode from 16 to 0, `actual_iv_size` could be zero. I prefer to avoid doing any operation in this case

Reply:
Ok, I understand what you mean. Do you think that if the `IV == NULL && iv_len == 0`, then we should at least set `ctx->iv_size = 0`? For example, consider the rather odd situation that we have some code like:
```
mbedtls_cipher_init();
mbedtls_cipher_setup();
mbedtls_cipher_set_iv( ctx, iv, 16 );
// Do something (1)
mbedtls_cipher_set_iv( ctx, NULL, 0 );
// Do something (2)
```
In this case, it seems that the second call to `mbedtls_cipher_set_iv()` will not really have any effect on `ctx`.